src/utils/Rotator.ts

```diff
@@ -0,0 +1,544 @@
+   */
+  isHandler(el: HTMLElement) {
+    const { handler } = this;
+    return handler === el;
+  }
+
+  /**
+   * Returns the focused element
+   * @return {HTMLElement}
+   */
+  getFocusedEl() {
+    return this.el;
+  }
+
+  /**
+   * Returns the parent of the focused element
+   * @return {HTMLElement}
+   */
+  getParentEl() {
+    return this.el?.parentElement;
+  }
+
+  /**
+   * Returns documents
+   */
+  getDocumentEl() {
+    return [this.el!.ownerDocument, document];
+  }
+
+  /**
+   * Return element position
+   * @param  {HTMLElement} el
+   * @param  {Object} opts Custom options
+   * @return {Object}
+   */
+  getElementPos(el: HTMLElement, opts = {}) {
+    const { posFetcher } = this;
+    return posFetcher ? posFetcher(el, opts) : getBoundingRect(el);
+  }
+
+  /**
+   * Return element rotation
+   * @param   {HTMLElement} el
+   * @returns {number} rotation
+   */
+  getElementRotation(el: HTMLElement): number {
+    var rotation = window.getComputedStyle(el, null).getPropertyValue('rotate') ?? null;
+    return rotation ? Number(rotation.replace('deg', '')) : 0;
+  }
+
+  /**
+   * Focus rotator on the element, attaches handlers to it
+   * @param {HTMLElement} el
+   */
+  focus(el: HTMLElement) {
+    // Avoid focusing on already focused element
+    if (el && el === this.el) {
+      return;
+    }
+
+    this.el = el;
+    this.updateContainer({ forceShow: true });
+    on(this.getDocumentEl(), 'pointerdown', this.handleMouseDown);
+  }
+
+  /**
+   * Blur from element
+   */
+  blur() {
+    this.container!.style.display = 'none';
+
+    if (this.el) {
+      off(this.getDocumentEl(), 'pointerdown', this.handleMouseDown);
+      delete this.el;
+    }
+  }
+
+  /**
+   * Start rotating
+   * @param  {Event} e
+   */
+  start(ev: Event) {
+    const e = ev as PointerEvent;
+    // @ts-ignore Right or middel click
+    if (e.button !== 0) return;
+    e.preventDefault();
+    e.stopPropagation();
+    const el = this.el!;
+    const rotator = this;
+    const config = this.opts || {};
+    const attrName = 'data-' + config.prefix + 'handler';
+    const rectRotation = this.getElementRotation(el!);
+    const rect = this.getElementPos(el!, { avoidFrameZoom: true, });
+    const target = e.target as HTMLElement;
+    this.handlerAttr = target.getAttribute(attrName)!;
+
+    const mouseFetch = this.mousePosFetcher;
+    this.startPos = mouseFetch
+      ? mouseFetch(e)
+      : {
+          x: e.clientX,
+          y: e.clientY,
+        };
+
+    this.startDim = {
+      t: rect.top,
+      l: rect.left,
+      w: rect.width,
+      h: rect.height,
+      r: rectRotation,
+    };
+    this.rectDim = {
+      t: rect.top,
+      l: rect.left,
+      w: rect.width,
+      h: rect.height,
+      r: rectRotation,
+    };
+
+    const dims = this.getElementPos(el!, { avoidFrameZoom: true, avoidFrameOffset: true });
+    this.center = {
+      x: dims.left + (dims.width / 2),
+      y: dims.top + (dims.height / 2),
```

Comment:
Wat is het verschil tussen dims en rect?

Reply:
Dat dims geen rekening met de frame offset houdt - en dus relatief tot het frame is ipv tot de browser